pkg/config/config.go

```diff
@@ -212,6 +217,26 @@ func New(opts ...Option) (*Config, error) {
 		}
 	}
 
+	// log the resolved config paths
+	absoluteConfigPaths := make([]string, len(c.paths))
+	for i, path := range c.paths {
+		absoluteConfigPaths[i], err = filepath.Abs(path)
+		if err != nil {
+			absoluteConfigPaths[i] = path
+		}
+	}
+
+	// allow the users to set datadir to a path like ~/.bacalhau or ~/something/idk/whatever
+	// and expand the path for them
+	dataDirPath := c.base.GetString(types.DataDirKey)
+	if dataDirPath != "" {
+		dataDirPath, err := ExpandPath(dataDirPath)
+		if err != nil {
+			return nil, fmt.Errorf("%s invalid: %w", types.DataDirKey, err)
+		}
+		c.base.Set(types.DataDirKey, dataDirPath)
+	}
```

Comment:
do we need to expand? we do validate in line 138

Reply:
138 is validating the contents of the config file(s) passed, ensuring iff repo dir is present in the file its an absolute path. 

The logic here is expanding values provided via flags, envvars, and --config flags, expanding them on the users behalf.